plutus-tx-plugin/executables/profile/Main.hs

```diff
@@ -24,12 +41,117 @@ import qualified PlutusCore.Default        as PLC
 
 import           Control.Lens.Combinators  (_2)
 import           Control.Lens.Getter       (view)
+import           Data.List                 (intercalate)
+import           Data.Maybe                (fromJust)
 import           Data.Proxy                (Proxy (Proxy))
 import           Data.Text                 (Text)
+import           Data.Time.Clock           (NominalDiffTime, UTCTime, diffUTCTime)
 import           Prettyprinter.Internal    (pretty)
 import           Prettyprinter.Render.Text (hPutDoc)
 import           System.IO                 (IOMode (WriteMode), withFile)
 
+data Stacks
+  = MkStacks
+  { -- | The variable name.
+    varName           :: String,
+    -- | The time when it starts to be evaluated.
+    startTime         :: UTCTime,
+    -- | The time spent on evaluating the functions it called.
+    timeSpentCalledFn :: NominalDiffTime
+  }
+  deriving (Show)
+
+-- | Write the time log of a program to a file in
+-- the plutus-tx-plugin/executables/profile/ directory.
+writeLogToFile ::
+  ToUPlc a PLC.DefaultUni PLC.DefaultFun =>
+  -- | Name of the file you want to save it as.
+  FilePath ->
+  -- | The program to be profiled.
+  [a] ->
+  IO ()
+writeLogToFile fileName values = do
+  let filePath = "plutus-tx-plugin/executables/profile/"<>fileName<>".timelog"
+  log <- pretty . view _2 <$> (rethrow $ runUPlcProfileExec values)
+  withFile
+    filePath
+    WriteMode
+    (\h -> hPutDoc h log)
+  processed <- processLog filePath
+  writeFile (filePath<>".stacks") processed
+  pure ()
+
+processLog :: FilePath -> IO [Char]
+processLog file = do
+  content <- readFile file
```

Comment:
Also, I'd split this into a pure part and an impure part. At the moment the whole thing is in IO, even though the only IO bit is reading the file at the beginning. Then we can write some tests for the pure bit!